### PR TITLE
GH-34812: [Packaging][Python] Use self-hosted arm64 Linux runner instead of Travis CI for Linux arm64 wheels

### LINKED ISSUE #34812: [Packaging][Python] Use self-hosted arm64 runner instead of Travis CI for arm64 wheels
### Describe the enhancement requested

We can use self-hosted arm64 runner in Crossbow jobs (thanks to Voltron Data!) by `runs-on: ["self-hosted", "Linux", "arm64"]`.
See also: #34717

There is only one Crossbow job that uses Travis CI. It's a job for wheels: https://github.com/apache/arrow/blob/fa4b17057570530bba913b92daeba186acb13cba/dev/tasks/tasks.yml#L460

We can use self-hosted arm64 runner for it. We can move away from Travis CI for Crossbow jobs.

Note that we still depend on Travis CI for apache/arrow CI jobs. See also: #20496

### Component(s)

Continuous Integration, Packaging, Python

### PR DESCRIPTION
**Pull request by eitsupi**
### Rationale for this change

Change one of the jobs that uses Travis CI to run on GitHub Actions self-hosted runner, reducing the number of Travis CI jobs.

### What changes are included in this PR?

Changes to use self-hosted arm64 linux runner for Crossbow jobs building wheels.

### Are these changes tested?

Yes

### Are there any user-facing changes?

No
* Closes: #34812

### CODE REVIEW

**Comment by eitsupi** [comment]
@github-actions crossbow submit wheel-manylinux*

**Comment by eitsupi** [comment]
`actions/setup-python@v4` is used in the following line, which does not appear to currently support arm64 platforms (actions/setup-python#108)
https://github.com/apache/arrow/blob/7e19111f2f81a4c0a8b3465f04a6b9f3ba4a06f1/dev/tasks/macros.jinja#L56-L62

We may need to use `ci/scripts/install_python.sh` instead.

**Comment by eitsupi** [comment]
@github-actions crossbow submit wheel-manylinux2014-cp39-arm64

**Comment by eitsupi** [comment]
@github-actions crossbow submit wheel-manylinux2014-cp39-arm64

**Comment by eitsupi** [comment]
@github-actions crossbow submit wheel-manylinux2014-cp39-arm64

**Comment by eitsupi** [comment]
@github-actions crossbow submit wheel-manylinux2014-cp39-arm64

**Comment by eitsupi** [comment]
It seems the job should be updated to use `quay.io/pypa/manylinux2014_aarch64:2022-10-25-fbea779`

```sh
$ docker buildx imagetools inspect quay.io/pypa/manylinux2014_arm64:2022-10-25-fbea779
ERROR: pulling from host quay.io failed with status code [manifests 2022-10-25-fbea779]: 401 Unauthorized
$ docker buildx imagetools inspect quay.io/pypa/manylinux2014_aarch64:2022-10-25-fbea779
Name:      quay.io/pypa/manylinux2014_aarch64:2022-10-25-fbea779
MediaType: application/vnd.docker.distribution.manifest.v2+json
Digest:    sha256:d6f5f8ef05bf539bb08971acb844a37246b6d89d32b86a4993f7f779699e18f4
```

**Comment by eitsupi** [comment]
@github-actions crossbow submit wheel-manylinux2014-cp39-arm64

**Comment by eitsupi** [comment]
@github-actions crossbow submit wheel-manylinux*

**Comment by eitsupi** [comment]
It seems working now.

Should some of the jinja macros be removed in this PR as they seem to be used only in the file that will be removed in this PR?
https://github.com/apache/arrow/blob/7e19111f2f81a4c0a8b3465f04a6b9f3ba4a06f1/dev/tasks/macros.jinja#L185-L239

File: dev/tasks/python-wheels/github.linux.yml [hunk 1]
```diff
@@ -22,15 +22,37 @@
...
+        run: echo "$HOME/.local/bin" >>"$GITHUB_PATH"
+      - name: Install python # Workaround for arm64 https://github.com/actions/setup-python/issues/108
+        shell: bash
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y python3-pip
+          pip3 install -U pip
+      - name: Install Archery
+        shell: bash
+        run: pip3 install -e arrow/dev/archery[all]
```
Comment (kou) [inline dev/tasks/python-wheels/github.linux.yml:54]: Could you do this in `macros.jinja` like https://github.com/apache/arrow/pull/34717/files#diff-77e693ebc06aba9fdd8c4b8d27e8ddcf54d91db8dbb7c060910e4a9c25d76a3c ?

File: dev/tasks/python-wheels/github.linux.yml [hunk 2]
```diff
@@ -49,6 +71,7 @@ jobs:
           archery docker run python-wheel-manylinux-test-imports
           archery docker run python-wheel-manylinux-test-unittests
 
+      {% if arch == "amd64" %}
```
Comment (kou) [inline dev/tasks/python-wheels/github.linux.yml:74]: Can we enable this on arm64 too?

File: dev/tasks/tasks.yml [hunk 3]
```diff
@@ -456,12 +456,13 @@ tasks:
 
 {############################## Wheel Linux ##################################}
 
-{% for ci, arch, arch_alias, x_y, manylinux in [("github", "amd64", "x86_64", "2_17", "2014"),
-                                                ("travis", "arm64", "aarch64", "2_17", "2014")] %}
+{% for arch, arch_alias, x_y, manylinux in [("amd64", "x86_64", "2_17", "2014"),
+                                                ("arm64", "aarch64", "2_17", "2014")] %}
```
Comment (kou) [inline dev/tasks/tasks.yml:460]:
```suggestion
                                             ("arm64", "aarch64", "2_17", "2014")] %}
```

**Review by kou — COMMENTED** [review]
> Should some of the jinja macros be removed in this PR as they seem to be used only in the file that will be removed in this PR?

Yes, please.

File: dev/tasks/python-wheels/github.linux.yml (hunk 2 above)
Comment (eitsupi) [reply dev/tasks/python-wheels/github.linux.yml]: I did not enable these in arm64 because Travis job did not seem having this, but I will try it out.

**Comment by eitsupi** [comment]
@github-actions crossbow submit -g wheel

**Review by kou — APPROVED** [review]
+1

**Pull request merged**